students/BrianB/lesson03/mailroom1.py

```diff
@@ -0,0 +1,154 @@
+                                                            an equal opportunity
+                                                            employer
+                                '''.format(donor[0], donor[1][-1])))
+                break  # this does not break out of [thank =_you()]
+        else:
+            print(name, ": There's a new donor!")
+            amt = input("New donor found! Please enter a donation amount:")
+            donor_data.append((name, [float(amt)]))
+            print(donor_data)
+            # print(gen_letter(donor))
+            print(dedent('''
+
+                            Dear {}!
+                            Thank for your donation of ${:,.2f}! We appreciate
+                            your contribution and without you nothing would
+                            be possible!
+
+                                                     Sincerely,
+                                                        Street Fighter, LLC
+                                                            an equal opportunity
+                                                            employer
+                                '''.format(name, float(amt))))
+            break
+
+
+def gen_stats(donor):
+    donations = donor[1]
+    total = sum(donations)
+    num = len(donations)
+    stats = (donor[0], total, num, total/num)
+    return stats
+
+
+def gen_letter():
+    return dedent('''
+
+                Dear {}!
+                     Thank for your donation of {:,.2f}! We appreciate
+                     your contribution and without you nothing would
+                     be possible!
+
+                                             Sincerely,
+                                                Street Fighter, LLC
+                                                    an equal opportunity
+                                                    employer
+                    '''.format(name, float(amt)))
+    pass
+
+
+def sort_key(donor):
+    return donor[1][2]
+
+
+print(sorted(donor_data, key=sort_key))
+
+
+def report():
+    print("in report")
+
+
+def quit():
+    print("quitting")
+    sys.exit()
+
+
+def main_menu():
+
+    while True:
+        answer = input(("What would you like to do?\n"
+                        "Pick One:\n"
+                        "1 - Send a Thank You\n"
+                        "2 - Create a Report\n"
+                        "3 - Quit\n"
+                        ">>> "))
+        print("you replied:", answer)
+        answer = answer.strip()
+        if answer == "1":
+            thank_you()
+        elif answer == "2":
+            report()
+        elif answer == "3":
+            quit()
+        else:
+            print("Please answer 1, 2, or 3")
+
+
+if __name__ == "__main__":
+    print("Welcome to the Mailroom!")
+
+    main_menu()
+
+    donor = ("Barney Rubble", [101, 201, 51])
+    assert gen_stats(donor) == ("Barney Rubble", 353, 117.67)
+
```

Comment:
looks like you still need the report part done. 

You can add that when you add dicts ...